src/ansys/geometry/core/designer/design.py

```diff
@@ -581,6 +581,45 @@ def delete_beam_profile(self, beam_profile: Union[BeamProfile, str]) -> None:
                 + " Ignoring request."
             )
 
+    @protect_grpc
+    @check_input_types
+    @ensure_design_is_active
+    def insert_file(self, file_location: Union[Path, str]) -> None:
+        """
+        Insert a file into the design.
+
+        Parameters
+        ----------
+        file_location : Union[~pathlib.Path, str]
+            Location on disk where the file is located.
+        """
+        # Upload the file to the server
+        filepath_server = self._modeler._upload_file(file_location)
+
+        # Insert the file into the design
+        self._design_stub.Insert(InsertRequest(filepath=filepath_server))
+        self._grpc_client.log.debug(f"File {file_location} successfully inserted into design.")
+
+        # Get a temporal design object to update the current one
+        tmp_design = Design("", self._modeler, read_existing_design=True)
+
+        # Update the design's components - add the new one
+        #
+        # If the list is empty, add the components from the new design
+        if not self._components:
+            self._components.extend(tmp_design.components)
+        else:
+            # Insert operation adds the inserted file as a component to the design.
+            for tmp_component in tmp_design.components:
+                # Otherwise, check which is the new component added
+                for component in self._components:
+                    if component.id == tmp_component.id:
+                        break
+                    # If not equal, add the component - since it has not been found
+                    self._components.append(tmp_component)
```

Comment:
I'll probably need to update the reference to parent component inside the new component

Reply:
Done!